plutus-metatheory/src/Builtin.lagda.md

```diff
@@ -79,7 +95,127 @@ data Builtin : Set where
+    a : ∀{n} → suc n ⊢♯
+    a = ` Z
+
+    b : ∀{n} → suc (suc n) ⊢♯
+    b = ` (S Z)
+
+   -- syntax for writing functions of 1,2,3, and 6 arguments
+    _[_]⟶_ : (n : ℕ) → n ⊢♯ → n ⊢♯ → Sig
+    n [ t ]⟶ r = sig n [ t ] r
+
+    _[_,_]⟶_ : (n : ℕ) → n ⊢♯ → n ⊢♯ → n ⊢♯ → Sig
+    n [ t , u ]⟶ r = sig n (t ∷⁺ [ u ]) r 
+
+    _[_,_,_]⟶_ : (n : ℕ) → n ⊢♯ → n ⊢♯ → n ⊢♯ → n ⊢♯ → Sig
+    n [ t , u , v ]⟶ r = sig n (t ∷⁺ u  ∷⁺ [ v ]) r 
+
+    _[_,_,_,_,_,_]⟶_ : (n : ℕ) → n ⊢♯ → n ⊢♯ → n ⊢♯ → n ⊢♯ → n ⊢♯ → n ⊢♯ → n ⊢♯ → Sig
+    n [ t , u , v , w , x , y ]⟶ r = sig n (t ∷⁺ u  ∷⁺ v ∷⁺ w  ∷⁺ x  ∷⁺ [ y ]) r
+    ```
+
+    The signature of each builtin
+
+    ```
+    signature : Builtin → Sig
+    signature addInteger                      = ∙ [ integer , integer ]⟶ integer
+    signature subtractInteger                 = ∙ [ integer , integer ]⟶ integer
+    signature multiplyInteger                 = ∙ [ integer , integer ]⟶ integer
+    signature divideInteger                   = ∙ [ integer , integer ]⟶ integer
+    signature quotientInteger                 = ∙ [ integer , integer ]⟶ integer
+    signature remainderInteger                = ∙ [ integer , integer ]⟶ integer
+    signature modInteger                      = ∙ [ integer , integer ]⟶ integer
+    signature equalsInteger                   = ∙ [ integer , integer ]⟶ bool
+    signature lessThanInteger                 = ∙ [ integer , integer ]⟶ bool
+    signature lessThanEqualsInteger           = ∙ [ integer , integer ]⟶ bool
+    signature appendByteString                = ∙ [ bytestring , bytestring ]⟶ bytestring
+    signature consByteString                  = ∙ [ integer , bytestring ]⟶ bytestring
+    signature sliceByteString                 = ∙ [ integer , integer , bytestring ]⟶ bytestring                            
+    signature lengthOfByteString              = ∙ [ bytestring ]⟶ integer
+    signature indexByteString                 = ∙ [ bytestring , integer ]⟶ integer
+    signature equalsByteString                = ∙ [ bytestring , bytestring ]⟶ bool
+    signature lessThanByteString              = ∙ [ bytestring , bytestring ]⟶ bool
+    signature lessThanEqualsByteString        = ∙ [ bytestring , bytestring ]⟶ bool
+    signature sha2-256                        = ∙ [ bytestring ]⟶ bytestring
+    signature sha3-256                        = ∙ [ bytestring ]⟶ bytestring
+    signature blake2b-256                     = ∙ [ bytestring ]⟶ bytestring
+    signature verifyEd25519Signature          = ∙ [ bytestring , bytestring , bytestring ]⟶ bool
+    signature verifyEcdsaSecp256k1Signature   = ∙ [ bytestring , bytestring , bytestring ]⟶ bool
+    signature verifySchnorrSecp256k1Signature = ∙ [ bytestring , bytestring , bytestring ]⟶ bool
+    signature appendString                    = ∙ [ string , string ]⟶ string
+    signature equalsString                    = ∙ [ string , string ]⟶ bool
+    signature encodeUtf8                      = ∙ [ string ]⟶ bytestring
```

Comment:
Would it be worth renaming `string` to `text`? We use `Text` on the Haskell side. We used to use `String`, but it's long gone.